Makefile

```diff
@@ -52,21 +55,35 @@ ROOT=$(shell pwd)
 
 .PHONY: lint proj program
 
-all: $(LIB_DIR)/$(DEVICE_LIBRARY) $(MSLIB_DIR)/mslib.a lint project
+all: $(APP_DEPS) lint project
 
-include $(MSLIB_DIR)/mslib.mk
-include $(DEV_DIR)/cfg.mk
+include device/$(DEVICE_FAMILY)/device_config.mk
+include $(foreach path,$(dir $(DEPS)),$(path)rules.mk)  
```

Comment:
I haven't tested it, but you might be able to use something like `$(LIB_DIR)_FOO` for the library level makefiles by replacing this with `$(foreach path,$(dir $(DEPS)),$(eval LIB_DIR := $(path)) $(eval include $(path)rules.mk))`